File: make_json.py
import ast
import json
import re

with open('data_pipeline/fetch_and_analyze.py', 'r', encoding='utf-8') as f:
    code = f.read()
    
match = re.search(r'SECTOR_MAP_FALLBACK\s*=\s*(\{[\s\S]*?\n\})', code)
if match:
    dict_str = match.group(1)
    d = ast.literal_eval(dict_str)
    with open('custom_sectors.json', 'w', encoding='utf-8') as out:
        json.dump(d, out, ensure_ascii=False, indent=4)
    print("Success")

File: make_json.js
const fs = require('fs');
const code = fs.readFileSync('data_pipeline/fetch_and_analyze.py', 'utf8');
const match = code.match(/SECTOR_MAP_FALLBACK\s*=\s*(\{[\s\S]*?\n\})/);
if (match) {
    let dictStr = match[1];
    dictStr = dictStr.replace(/'/g, '"');
    fs.writeFileSync('custom_sectors.json', dictStr);
    console.log('Created custom_sectors.json');
} else {
    console.log('Not found');
}
